### PR TITLE
Tmpdir cleanup

### PR DESCRIPTION
**Pull request by geaaru**

### CODE REVIEW

**Comment by mudler** [comment]
looks good! wdyt about adding a simple (integration) test?

File: pkg/compiler/backend/simpledocker.go [hunk 1]
```diff
@@ -237,7 +237,7 @@ func (*SimpleDocker) ExtractRootfs(opts compiler.CompilerBackendOptions, keepPer
 // ]
 // Changes uses container-diff (https://github.com/GoogleContainerTools/container-diff) for retrieving out layer diffs
 func (*SimpleDocker) Changes(fromImage, toImage string) ([]compiler.ArtifactLayer, error) {
```
Comment (mudler) [inline pkg/compiler/backend/simpledocker.go:239]: Probably the same applies for the img backend

File: cmd/root.go [hunk 2]
```diff
@@ -61,6 +61,20 @@ var RootCmd = &cobra.Command{
...
+		// needed.
+		err = config.LuetCfg.GetSystem().InitTmpDir()
+		if err != nil {
+			Fatal("failed on init tmp basedir:", err.Error())
+		}
+	},
+	PersistentPostRun: func(cmd *cobra.Command, args []string) {
+		// Cleanup all tmp directories used by luet
+		err := config.LuetCfg.GetSystem().CleanupTmpDir()
+		if err != nil {
```
Comment (mudler) [inline cmd/root.go:75]: I have a bit of mixed feelings about making this fatal, but anyway it's being run after all was already being run....

Comment (mudler) [reply cmd/root.go:75]: Maybe make it just a warning?

File: pkg/compiler/backend/simpledocker.go (hunk 1 above)
Comment (geaaru) [reply pkg/compiler/backend/simpledocker.go:239]: > looks good! wdyt about adding a simple (integration) test?

Not so easy... tests are executed in parallel and if I clean up the directory I think that I broke other tests. I will integrate only the test to create a temp file and a temporary dir.

File: cmd/root.go (hunk 2 above)
Comment (geaaru) [reply cmd/root.go:75]: done

File: pkg/compiler/backend/simpledocker.go (hunk 1 above)
Comment (mudler) [reply pkg/compiler/backend/simpledocker.go:239]: Integration tests aren't run in parallel, iirc not even the unit tests

**Pull request merged**